webapp/components/Modal/ReportModal.vue

```diff
@@ -8,14 +8,36 @@
 
     <!-- eslint-disable-next-line vue/no-v-html -->
     <p v-html="message" />
-
-    <template slot="footer">
-      <ds-button class="cancel" icon="close" @click="cancel">{{ $t('report.cancel') }}</ds-button>
+    <ds-radio
+      v-model="form.reasonCategory"
+      :schema="formSchema.reasonCategory"
+      :label="$t('report.reason.category.label')"
+      :options="form.reasonCategoryOptions"
+      labelProp="label"
+    />
+    <ds-input
+      class="reason-description"
+      v-model="form.reasonDescription"
+      :schema="formSchema.reasonDescription"
+      :label="$t('report.reason.description.label')"
+      :placeholder="$t('report.reason.description.placeholder')"
+      type="textarea"
+      rows="5"
+    />
+    <small class="smallTag">
+      {{ form.reasonDescription.length }}/{{ formSchema.reasonDescription.max }}
+    </small>
+    <ds-space />
+    <template #footer>
+      <ds-button class="cancel" icon="close" @click="cancel">
+        {{ $t('report.cancel') }}
+      </ds-button>
 
       <ds-button
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-14T19:42:29Z" title="Monday, October 14th 2019, 9:42:29 pm +02:00">Oct 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -9,13 +9,35 @@
     <!-- eslint-disable-next-line vue/no-v-html -->
     <p v-html="message" />
 
+    <ds-radio
+      :value="form.reasonCategory"
+      :schema="formSchema.reasonCategory"
+      :label="$t('report.reason.category.label')"
+      :options="form.reasonCategoryOptions"
+      labelProp="label"
+    />
+    <ds-input
+      class="reason-description"
+      :value="form.reasonDescription"
+      :schema="formSchema.reasonDescription"
+      :label="$t('report.reason.description.label')"
+      :placeholder="$t('report.reason.description.placeholder')"
+      type="textarea"
+      rows="5"
+    />
+    <small class="smallTag">
+      {{ form.reasonDescription.length }}/{{ formSchema.reasonDescription.max }}
+    </small>
+    <ds-space />
+
     <template slot="footer">
       <ds-button class="cancel" icon="close" @click="cancel">{{ $t('report.cancel') }}</ds-button>
 
       <ds-button
         danger
         class="confirm"
         icon="exclamation-circle"
+        :disabled="failsValidations"
```


This is better solved with the `{ error }` slot-scope, look here: https://styleguide.human-connection.org/data-input/dsform under "Advanced usage / Validation".

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-15T17:58:15Z" title="Tuesday, October 15th 2019, 7:58:15 pm +02:00">Oct 15, 2019</time>_

---


in this case, difficult to achieve because we have a `<template #footer>` wrapping the `cancel/report` buttons... you cannot use `<template slot-scope="{ errors }">` with nested `templates`, but if we convert this to a `<div slot="footer">`, it completely changes the `styling` of the buttons to a `green` confirm button, instead of the desired `red` report button